### PR TITLE
Fix crash on saving config when directory doesn't exist

### PR DESCRIPTION
**Pull request by Buckminsterfullerene02**
When doing any save action it was crashing on the no such directory exists error.

### CODE REVIEW

**Comment by trumank** [comment]
I'm not able to reproduce this and I'm hesitant to to just slap a `create_dir_all` on top of it. Can you reliably reproduce the issue if you remove the config and cache directories?

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /target
 /hook/target
+/.idea
```
Comment (jieyouxu) [inline .gitignore:3]: Nit: this should probably go into your local `~/.gitconfig_global` or something like that (also missing newline).

**Review by jieyouxu — APPROVED** [review]
The fix looks correct to me: I think `fs::create_dir` originally worked locally for me because `~/.config` etc. already existed for me, so it didn't fail to create a dir when its parent didn't exist.

File: .gitignore (hunk 1 above)
Comment (trumank) [reply .gitignore:3]: Adding common build tools to `.gitignore` is fine

**Pull request merged**